### ISSUE TITLE
HTML report when running tests from command line

### ISSUE THREAD

**Issue by heidivanparys** [opengeospatial/ets-ogcapi-features10#107]
I think it would be useful if a HTML report would be generated when running the tests from the command line. I now did a quick fix by changing `false` to `true` on line 117 in `org.opengis.cite.ogcapifeatures10.TestNGController`.

```
this.executor = new TestNGExecutor(tngSuite.toString(), resultsDir.getAbsolutePath(), true);
```

Are there any other ways to enable the generation of a HTML report? Or can the code be changed to incorporate this feature?

**Comment by dstenger** [comment]
Thank you for reporting.
Can you please describe how you executed the tests? Did you use an IDE and started the tests via TestNGController or did you use the shell to run the tests?

**Comment by heidivanparys** [comment]
I used the shell to run the tests. So in order to get a HTML report, I changed the code, executed `mvn package` and executed the command as described on [How to run the tests](https://github.com/opengeospatial/ets-ogcapi-features10/blob/master/src/site/asciidoc/how-to-run-the-tests.adoc).

**Comment by dstenger** [comment]
@keshav-nangare Can you please take a look at this problem?

**Comment by keshav-nangare** [comment]
@dstenger 

Following is the way we can provide runtime argument for generating HTML report:

1. We will define a new argument in the CommandLineArguments class(e.g outputDir) to accept a parameter for HTML report generation.
2. We need to make changes in the main method of TestNGController, while passing arguments to the TestNGController constructor which will set the runtime parameter.
```
Before change: 
        TestNGController controller = new TestNGController(testRunArgs.getOutputDir());
        public TestNGController(String outputDir) 

After change: 
        TestNGController controller = new TestNGController(testRunArgs.getOutputDir(), testRunArgs.defaultListner());
        public TestNGController(String outputDir, boolean defaultListner) 
```

3. Also, we need to change in CTL script as the definition of the TestNGController constructor changed. The CTL script is using the constructor to create object so we need to update with the new parameter.
```
Before change: <xsl:variable name="controller" select="tng:new($outputDir)" />
After change:  <xsl:variable name="controller" select="tng:new($outputDir, false)" />
```

When we set the value to true the following HTML report gets generated.
[result.zip](https://github.com/opengeospatial/ets-ogcapi-features10/files/4456402/result.zip)

What is your thought on this proposed solution?

**Comment by dstenger** [comment]
Please also test this workflow with aio.jar, document the result and if it works fine, please create a pull request.

**Comment by keshav-nangare** [comment]
@dstenger 

I have tested workflow with aio.jar and it worked fine. I will create PR for the same.
Please find below the test execution log:
```
cmd>java -jar ets-ogcapi-features10-0.6-SNAPSHOT-aio.jar -o ../ -l true classes/test-run-props.xml

Request method: GET
Request URI:    http://www.w3schools.com/xml/note.xml/
Proxy:                  <none>
Request params: <none>
Query params:   <none>
Form params:    <none>
Path params:    <none>
Headers:                Accept=application/json, application/javascript, text/javascript, text/json
Cookies:                <none>
Multiparts:             <none>
Body:                   <none>
Request method: GET
Request URI:    http://www.w3schools.com/xml/note.xml/
Proxy:                  <none>
Request params: <none>
Query params:   <none>
Form params:    <none>
Path params:    <none>
Headers:                Accept=*/*
Cookies:                <none>
Multiparts:             <none>
Body:                   <none>
Request method: GET
Request URI:    http://www.w3schools.com/xml/note.xml
Proxy:                  <none>
Request params: <none>
Query params:   <none>
Form params:    <none>
Path params:    <none>
Headers:                Accept=application/json, application/javascript, text/javascript, text/json
Cookies:                <none>
Multiparts:             <none>
Body:                   <none>
Test results: file:/Users/keshavnangare/ogc/test-suite/ets-ogcapi-features10/target/../testng/0f2867b1-e097-4c2d-85c6-ac02b90a98b5/testng-results.xml
```

**Comment by keshav-nangare** [comment]
Fixed with #109 PR.
